.travis.yml

```diff
@@ -14,20 +14,29 @@ jobs:
       services:
         - xvfb
       before_install:
-        - mkdir qunit
-        - wget -O qunit/qunit-1.18.0.css http://code.jquery.com/qunit/qunit-1.18.0.css
-        - wget -O qunit/qunit-1.18.0.js  http://code.jquery.com/qunit/qunit-1.18.0.js
+        - sudo add-apt-repository -y ppa:deadsnakes/ppa
+        - sudo apt-get -q update
+        - sudo apt-get -y install python3.10
+        - python --version
         - npm install testem
         - ./node_modules/.bin/testem launchers
 
+      before_script:
+        - (cd scripts && python3.10 make_doc.py)
+        - (cd scripts && python3.10 transform_full_grammar.py)
+        - (cd scripts && python3.10 make_dist.py)
+
       script:
-        - ./node_modules/.bin/testem --launch "Firefox,Headless Chrome" -t www/tests/qunit/run_tests.html ci
+        # Some day, put back Chrome or "Headless Chrome" as well. For some reason
+        # as of 2022 Aug 23, the testem websocket connection is sometimes reset,
+        # which prevents the browser from reporting results to the testem process
+        # when running on Chrome.
+        - ./node_modules/.bin/testem --launch Firefox -t www/tests/run_tests.html ci
 
     - language: python
-      python: 3.8
+      python: 3.10
       before_install: pip install --upgrade pip
       before_script: pip install flake8
       script:
         - EXCLUDE=./.*,www/src/Lib/test/badsyntax_3131.py
-        # stop the build if there are Python syntax errors
-        - flake8 . --exclude=$EXCLUDE --exit-zero --select=E999 --show-source --statistics
+        - flake8 . --exclude=$EXCLUDE --select=E9,F63,F7,F82 --show-source --statistics
```

Comment:
```suggestion
        - flake8 . --exclude=$EXCLUDE --exit-zero --select=E9,F63,F7,F82 --show-source --statistics
```

Reply:
The `undefined names` are basically saying if Python was a compiled language then this code would not compile.  For some _magical_ variables that brython always creates, you can do `flake --builtins=__BRYTHON__,JSObject` but it is safer to avoid sweeping too much under the carpet with this technique.